pkg/controller/healthcheck/add.go

```diff
@@ -46,7 +45,7 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
 				HealthCheck:   general.CheckManagedResource(constants.ManagedResourceNamesSeed),
 			},
 		},
-		sets.New(gardencorev1beta1.ShootSystemComponentsHealthy), // TODO(vpnachev), remove this condition removal in a future version.
+		nil,
```

Comment:
```suggestion
		sets.New(),
```

Wouldn't it be better to pass an empty set here so that we guarantee that this argument does not change to something unexpected in the future?

Reply:
You mean if this argument of `DefaultRegistration` changes to something other than `sets.Set[gardencorev1beta1.ConditionType])` that still accepts nil as parameter?

Reply:
Yes exactly, basically if `DefaultRegistration` suddenly accepts different argument that can be represented by `nil`